run/exec.go

```diff
@@ -59,12 +77,10 @@ func Exec(
 
 	results := startCmdRunner(cmds)
 
-	errs := errors.L()
-
 	for _, stack := range stacks {
 		cmd := exec.Command(cmd[0], cmd[1:]...)
 		cmd.Dir = stack.HostPath()
-		cmd.Env = os.Environ()
+		cmd.Env = append(os.Environ(), stackEnvs[stack.Path()]...)
```

Comment:
I think we should handle the de-duplication ourselves (maybe in the future) and log the overridden entries, otherwise, it could be very confusing to understand some scenarios later when we introduce `config.run` per-directory (hierarchy, etc).
Even today, if the user us unaware of all envs set in the project and try to just export something in the shell and run `terramate run`, his value will be ignored and no log will show this.
But I think it's minor at this point.

Reply:
Agreed, needs improvement :100:

Reply:
I also dug into the Go's deduplication and it changed to resemble shells in Go1.9, before that it used the first entry... we already depend on go1.17 so it's ok. Just sharing this in case you maybe knows anything else that could be a problem in this case. Feel free to ignore.

Reply:
https://github.com/golang/go/issues/19877

Reply:
I was not aware of that, just tested and checked that the current behavior is to use the last one (tests also cover that, if behavior changes it would fail). Thanks for checking this :100: